### PR TITLE
Adding pushdown usage markdown file

### PR DESCRIPTION
**Pull request by suryasoma**

### CODE REVIEW

File: PUSHDOWN.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+**Note:** that this will enable query pushdown on the entire Spark session. If you want to disable pushdown, you will need to call BigQueryConnectorUtils.disablePushdownSession
+
+#### Python code for disabling Query Pushdown
+```
+# Disable pushdown
+
+from google.cloud.spark.bigquery import big_query_connector_utils
+big_query_connector_utils.disablePushdownSession(spark)
+```
+**Note:** If the above code gives ModuleNotFoundError, you can directly disable pushdown for the Spark session as `spark.sparkContext._jvm.com.google.cloud.spark.bigquery.BigQueryConnectorUtils.enablePushdownSession(spark._jsparkSession)`
```
Comment (davidrabinowitz) [inline PUSHDOWN.md:59]: Line is redundant (line 48)

File: PUSHDOWN.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+| Sorts (ORDER BY)                                                           | [ORDER BY](https://spark.apache.org/docs/3.3.0/sql-ref-syntax-qry-select-orderby.html) and [SORT BY](https://spark.apache.org/docs/3.3.0/sql-ref-syntax-qry-select-sortby.html) clauses |
+| Union and Union All                                                        | [UNION and UNION ALL clauses](https://spark.apache.org/docs/3.3.0/sql-ref-syntax-qry-select-setops.html#union)                                                                          |
+| Window functions and Windowing clauses                                     | [WINDOW clause](https://spark.apache.org/docs/3.3.0/sql-ref-syntax-qry-select-window.html#content)                                                                                      |
+
+
+## Caveats/Limitations
+
+* Because the translation in Query Pushdown from the Spark LogicalPlan to SQL requires almost a one-to-one translation of Spark SQL operators to BigQuery expressions, not all of Spark SQL operators can be pushed down. When pushdown fails, the connector falls back to a less-optimized execution plan. The unsupported operations are instead performed in Spark.
```
Comment (davidrabinowitz) [inline PUSHDOWN.md:254]: s/when pushdown/when the pushdown/

Comment (davidrabinowitz) [inline PUSHDOWN.md:254]: s/falls back to a less-optimized execution plan/falls back to the original execution plan/

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: PUSHDOWN.md (hunk 2 above)
Comment (suryasoma) [reply PUSHDOWN.md]: Ack

Comment (suryasoma) [reply PUSHDOWN.md]: Ack

File: PUSHDOWN.md (hunk 1 above)
Comment (suryasoma) [reply PUSHDOWN.md]: Removed

**Review by VinayLondhe14 — APPROVED** [review]
Would be nice to have a section in the main README.md that points to this PUSHDOWN.md file

File: PUSHDOWN.md [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+| Min                 | [min](https://spark.apache.org/docs/latest/api/sql/index.html#min)                  |
+| StddevPop           | [stddev_pop](https://spark.apache.org/docs/latest/api/sql/index.html#stddev_pop)    |
+| StddevSamp          | [stddev_samp](https://spark.apache.org/docs/latest/api/sql/index.html#stddev_samp)  |
+| Sum                 | [sum](https://spark.apache.org/docs/latest/api/sql/index.html#sum)                  |
+| VariancePop         | [var_pop](https://spark.apache.org/docs/latest/api/sql/index.html#var_pop)          |
+| VarianceSamp        | [var_samp](https://spark.apache.org/docs/latest/api/sql/index.html#var_samp)        |
+
+### Boolean Functions
+
+| Spark function name   | Spark SQL function name                                                           |
```
Comment (davidrabinowitz) [inline PUSHDOWN.md:99]: Please change to the relevant operator (and - &&, contains - `LIKE '%foo%'` etc.)

File: PUSHDOWN.md [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+| Sqrt                 | [sqrt](https://spark.apache.org/docs/latest/api/sql/index.html#sqrt)         |
+| Tan                  | [tan](https://spark.apache.org/docs/latest/api/sql/index.html#tan)           |
+| Tanh                 | [tanh](https://spark.apache.org/docs/latest/api/sql/index.html#tanh)         |
+| IsNan                | [isnan](https://spark.apache.org/docs/latest/api/sql/index.html#isnan)       |
+| SigNum               | [signum](https://spark.apache.org/docs/latest/api/sql/index.html#signum)     |
+
+### Miscellaneous Operators
+
+| Spark function name    | Spark SQL function name                                                                                                                      |
+|:-----------------------|:---------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (davidrabinowitz) [inline PUSHDOWN.md:161]: Change to operator when relevant

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
Please change to spark function names and operators from the reference: https://spark.apache.org/docs/latest/sql-ref-functions.html

File: PUSHDOWN.md (hunk 3 above)
Comment (suryasoma) [reply PUSHDOWN.md]: Changed `Spark function name`  to `Spark Logical Plan Construct`

File: PUSHDOWN.md (hunk 4 above)
Comment (suryasoma) [reply PUSHDOWN.md]: Ack

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**